tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -83,5 +85,74 @@ private string TranslateError (string culture, string errorCode)
+		public void SetUp ()
+		{
+			commonIgnoreList = ReadFile ($"{Directory.GetCurrentDirectory ()}/TaskTests/LocalizationIgnore/common-Translations.ignore") as List<string>;
+		}
+
+		[TestCase ("cs-CZ")]
+		[TestCase ("de-DE")]
+		[TestCase ("es-ES")]
+		[TestCase ("fr-FR")]
+		[TestCase ("it-IT")]
+		[TestCase ("ja-JP")]
+		[TestCase ("ko-KR")]
+		[TestCase ("pl-PL")]
+		[TestCase ("pt-BR")]
+		[TestCase ("ru-RU")]
+		[TestCase ("tr-TR")]
+		[TestCase ("zh-CN")]
+		[TestCase ("zh-TW")]
+		public void AllErrorTranslation (string culture)
+		{
+			string errorList = string.Empty;
+			List<string> cultureIgnoreList = null;
+			List<string> commonValidEntries = new List<string> ();
+			List<string> cultureValidEntries = new List<string> ();
+
+			string fullCommonPath = $"{Directory.GetCurrentDirectory ()}/TaskTests/LocalizationIgnore/common-Translations.ignore";
+			string fullCulturePath = $"{Directory.GetCurrentDirectory ()}/TaskTests/LocalizationIgnore/{culture}-Translations.ignore";
+			string shortCommonPath = "xamarin-macios/tests/msbuild/Xamarin.MacDev.Tasks.Tests/Tasktests/LocalizationIgnore/common-Translations.ignore";
+			string shortCulturePath = $"xamarin-macios/tests/msbuild/Xamarin.MacDev.Tasks.Tests/Tasktests/LocalizationIgnore/{culture}-Translations.ignore";
+			CultureInfo originalCulture = Thread.CurrentThread.CurrentUICulture;
+
+			cultureIgnoreList = ReadFile (fullCulturePath) as List<string>;
+
+			foreach (var errorCodeInfo in typeof (MSBStrings).GetProperties ()) {
+				try {
+					var errorCode = errorCodeInfo.Name;
+					if (errorCode == "ResourceManager" || errorCode == "Culture")
+						continue;
+					string englishError = TranslateError ("en-US", errorCode);
+					string newCultureError = TranslateError (culture, errorCode);
+
+					if (commonIgnoreList.Contains (errorCode)) {
+						Assert.AreEqual (englishError, newCultureError, $"{errorCode} is translated. Remove {errorCode} from {shortCommonPath}");
+						commonValidEntries.Add (errorCode);
+					} else if (cultureIgnoreList.Contains (errorCode)) {
+						Assert.AreEqual (englishError, newCultureError, $"{errorCode} is translated. Remove {errorCode} from {shortCulturePath}");
+						cultureValidEntries.Add (errorCode);
+					} else if (englishError == newCultureError)
+						errorList += $"{errorCode} ";
+				} finally {
+					Thread.CurrentThread.CurrentUICulture = originalCulture;
+				}
+			}
+
+			Assert.AreEqual (string.Empty, errorList, $"The following errors were not translated. Add them to {shortCommonPath} or {shortCulturePath}");
+			Assert.IsEmpty (cultureIgnoreList.Except (cultureValidEntries), $"Not all error codes in {shortCulturePath} are valid or are repeated. Please remove.");
+			Assert.IsEmpty (commonIgnoreList.Except (commonValidEntries), $"Not all error codes in {shortCommonPath} are valid or are repeated. Please remove.");
+		}
+
+		IList<string> ReadFile (string path)
+		{
+			if (!File.Exists (path))
+				return new List<string> ();
```

Comment:
You can return an empty array if you do the other change to the field type at the top:

```suggestion
				return Array.Empty<string> ();
```

this is more efficient (it won't create a new instance every time this method is called).